### PR TITLE
Update flow to 0.54.1

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers cc: @mmaxim

### CODE REVIEW

**Comment by mmaxim** [comment]
Thanks!

File: shared/actions/signup.js [hunk 1]
```diff
@@ -35,16 +35,15 @@ function nextPhase() {
...
+    dispatch({payload: {}, type: Constants.startRequestInvite})
+    dispatch(nextPhase())
+  }
 }
 
 function checkInviteCode(inviteCode: string) {
-  return (dispatch: Dispatch) =>
-    new Promise((resolve, reject) => {
+  return (dispatch: Dispatch) => {
+    const p: Promise<*> = new Promise((resolve, reject) => {
```
Comment (chrisnojima) [inline shared/actions/signup.js]: not sure why flow wants this now but this fixes the errors

File: shared/common-adapters/icon.native.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ import type {IconType, Props} from './icon'
 
 // In order to optimize this commonly used component we use StyleSheet on all the default variants
 // so we can pass IDs around instead of full objects
+// $FlowIssue
```
Comment (MarcoPolo) [inline shared/common-adapters/icon.native.js]: :sheep: What's the issue?

Comment (chrisnojima) [reply shared/common-adapters/icon.native.js]: dunno, not happy about the maps i guess

**Pull request merged**